nio_cli/commands/blockcheck.py

```diff
@@ -0,0 +1,241 @@
+                                lines[block_indices[i]:block_indices[i+1]]:
+                            print('Add "{}" to the {} block'.format(
+                                key, lines[block_indices[i]]))
+                    except IndexError:
+                        if key not in lines[block_indices[i]:]:
+                            print('Add "{}" to the {} block'.format(
+                                key, lines[block_indices[i]]))
+        else:
+            print(
+                'Run `nio buildreadme` '
+                'as long as the spec.json file is complete'
+            )
+        print('')
+
+    def check_release(self):
+
+        if os.path.exists('release.json'):
+            with open('release.json') as f:
+                release_dict = json.load(f)
+
+        for block in self.blocks_in_spec:
+            if 'nio/' + block not in release_dict:
+                print('Add {} block to release.json'.format(block))
+            for key in ['url', 'version', 'language']:
+                if key not in release_dict['nio/' + block] \
+                        or release_dict['nio/' + block][key] == '':
+                    print('Add a {} to the {} block'.format(key, block))
+            self.release_versions_dict[block] = \
+                release_dict['nio/' + block]['version']
+        print('')
+
+    def check_version(self):
+
+        for block in self.blocks_in_spec:
+            if self.release_versions_dict[block] != \
+                    self.spec_versions_dict[block]:
+                print(
+                    'Spec.json and release.json versions do not match for '
+                    '{} block'.format(block)
+                )
+            if self.release_versions_dict[block] \
+                    not in self.file_versions_and_names_dict['versions']:
+                print(
+                    'The {} version in the release file '
+                    'does not match the version in its '
+                    'block file'.format(block)
+                )
+                if self.spec_versions_dict[block] \
+                    not in self.file_versions_and_names_dict['versions']:
+                    print(
+                        'The {} version in the spec file '
+                        'does not match the version in its '
+                        'block file'.format(block)
+                    )
+        print('')
+
+    def check_naming(self):
+
+        for block in self.blocks_in_spec:
+            if '_' in block:
+                print(
+                    '{} class name should be camelCased format'.format(block)
+                )
+            if block not in self.file_versions_and_names_dict['classes']:
+                print(
+                    '{} block needs to have a defined class'.format(block)
+                )
+
+        for block in self.block_files:
+            if not block.islower():
+                print(
+                    '{} file name should be lowercased and '
```

Comment:
I had originally checked for the `_block.py`, but felt it was redundant because `self.block_files` is generated (line 21) by finding `_block.py` in the file name.